pt/controllers/WithSunriseApplication.java

```diff
@@ -0,0 +1,102 @@
+package controllers;
+
+import basicauth.BasicAuthTestModule;
+import categorytree.CategoryTreeTestModule;
+import com.google.inject.Module;
+import com.google.inject.util.Modules;
+import common.controllers.TestableReverseRouter;
+import common.controllers.TestableSphereClient;
+import ctpclient.CtpClientTestModule;
+import inject.ApplicationTestModule;
+import inject.CtpModelsTestModule;
+import io.sphere.sdk.categories.Category;
+import io.sphere.sdk.categories.queries.CategoryQuery;
+import org.apache.commons.lang3.RandomStringUtils;
+import play.Application;
+import play.Configuration;
+import play.Environment;
+import play.Mode;
+import play.inject.guice.GuiceApplicationBuilder;
+import play.libs.ws.WS;
+import play.libs.ws.WSRequest;
+import play.mvc.Controller;
+import play.mvc.Http;
+import reverserouter.ReverseRouterTestModule;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static common.JsonUtils.readCtpObject;
+import static play.test.Helpers.running;
+import static play.test.Helpers.testServer;
+
+public abstract class WithSunriseApplication {
+    public static final int ALLOWED_TIMEOUT = 100000;
+
+    @FunctionalInterface
+    public interface CheckedConsumer<T> {
+
+        void apply(T t) throws Exception;
+    }
+
+    protected final void run(final Application app, final String url, final CheckedConsumer<WSRequest> test) {
+        running(testServer(3333, app), () -> {
+            try {
+                test.apply(WS.url("http://localhost:3333" + url));
+            } catch (Throwable t) {
```

Comment:
Not sure why I put here a Throwable when the CheckedConsumer throws an Exception. I'll change it.